### ISSUE TITLE
Don't hide nullability problems behind a cast

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#3484]
## Expected Behavior of the rule

Noncompliant:
```kotlin
fun foo(bar: Any?) {
  val x = bar as String
}
```

Compliant:
```kotlin
fun foo(bar: Any?) {
  val x = bar!! as String
}
```

## Context
The cast operator can hide null problems in your code. I propose a rule that forbid any cast from a nullable type to a non-null type. If you want to do that you first should cast from nullable to non-null and then do the cast as shown in the example.

The idea is to write code that show that it could fails for two things (nullability and type) and not just one (cast).

I had this idea reading #3479 so they are related but they are not the same.

**Comment by atulgpt** [comment]
Hi @BraisGabin I am willing to implement this issue, should I proceed and will your review the PR?

**Comment by BraisGabin** [comment]
Please, go ahead! I'll review it.

**Comment by atulgpt** [comment]
I @BraisGabin added the https://github.com/detekt/detekt/pull/5653
